File: backend/src/main/java/com/example/stockwatchlist/model/Stock.java
package com.example.stockwatchlist.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "stocks")
public class Stock {

    @Id
    private String id;
    private String ticker;
    private String company;
    private String marketCap;
    private Double psRatio;
    private String dailyChange;

    public Stock() {
    }

    public String getId() { return id; }
    public String getTicker() { return ticker; }
    public String getCompany() { return company; }
    public String getMarketCap() { return marketCap; }
    public Double getPsRatio() { return psRatio; }
    public String getDailyChange() { return dailyChange; }
}

File: frontend/lib/models/stock.dart
class Stock {
  const Stock({
    required this.ticker,
    required this.company,
    required this.marketCap,
    required this.psRatio,
    required this.dailyChange,
  });

  final String ticker;
  final String company;
  final String marketCap;
  final double psRatio;
  final String dailyChange;

  factory Stock.fromJson(Map<String, dynamic> json) {
    return Stock(
      ticker: json['ticker'] as String,
      company: json['company'] as String,
      marketCap: json['marketCap'] as String,
      psRatio: (json['psRatio'] as num).toDouble(),
      dailyChange: json['dailyChange'] as String,
    );
  }
}
